doc/api/n-api.md

```diff
@@ -1725,9 +1725,9 @@ Returns `napi_ok` if the API succeeded.
 This API creates a new reference with the specified reference count
 to the value passed in.
 
-In Node-API version 8 and earlier, a reference could only be created for
-object, function, external, and symbol value types. However, in newer Node-API
-versions, a reference can be created for any value type.
+In current Node-API version a reference can be created only for
```

Comment:
Maybe we shouldn't mix version-specific discussion with documentation that applies to all versions. Instead, maybe we should have headings for experimental and subsequent versions. Maybe like

-----------------------------
Default behaviour (currently version 8, which does not support basic types) goes here.

Change history:
- experimental: Supports all types, including basic types.
-----------------------------

When we release Node-API version 9, we can replace "experimental" with "Version 9" in the "Change history" section, and re-add the "experimental" bullet above "Version 9" if we ever change the behaviour again.

Reply:
I have updated the docs based on this suggestion and extracted the experimental behavior into its own paragraph.